### PR TITLE
Split: stability improvements over serial

### PR DESCRIPTION
**Pull request by dexter93**
## Description
Stability fixes over cases observed in edge-case scenarios ( small hardware FIFOs). Should offer more robust communication in split config.

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by dexter93** [comment]
note: timeout logic should probably be extended to `serial_transport_receive` as well. Still testing, will update.

File: quantum/split_common/split_util.c [hunk 1]
```diff
@@ -59,14 +59,15 @@ volatile bool isLeftHand = true;
...
         // This will return true if a USB connection has been established
         if (usb_connected_state()) {
-            return true;
+            is_active = true;
         }
         wait_ms(SPLIT_USB_TIMEOUT_POLL);
     }
-    return false;
+    return is_active;
 }
```
Comment (tzarc) [inline quantum/split_common/split_util.c:71]: This should be reverted -- we want it to exit early when connected. Your changes unconditionally continue to call the delay.

**Comment by tzarc** [comment]
You're going to have to do a bit more convincing -- I don't really see the point of the content in this PR at all.

File: quantum/split_common/split_util.c (hunk 1 above)
Comment (dexter93) [reply quantum/split_common/split_util.c:71]: > This should be reverted -- we want it to exit early when connected. Your changes unconditionally continue to call the delay.

Even if the master side exits early when USB connected, the slave is stuck in this delay loop. Behavior wise, it's the same, no split comms until the delay is over. However, this patch avoids the initial packet spam from master while the slave is still looking to see if it's USB is active ( no effective payload getting exchanged).

**Comment by dexter93** [comment]
> You're going to have to do a bit more convincing -- I don't really see the point of the content in this PR at all.

Without [#4a1b61e](https://github.com/qmk/qmk_firmware/pull/22233/commits/4a1b61eb8a12e91379484d3bfd51b4906c1de4bb) , it seems that hardware with small hw FIFOs experience timeouts faster than they should ( although packets have been sent properly). From `chnWriteTimeout` ref:
```* @brief   Channel blocking write with timeout.
 * @details The function writes data from a buffer to a channel. If the channel
 *          is not ready to accept data then the calling thread is suspended.
```
Which in turns means that FIFOs have to be empty/have available space for the outgoing payload ( adjusting `SERIAL_BUFFERS_SIZE` didn't seem to be reliable in my testing). This is especially tricky on hw with 1 byte wide FIFOs ( sn32, kinetis) or even UART hardware with no FIFOs - since the `#define`'d timeout period is shorter than the time needed for the data to be sent if the transfer is big enough. A small adjustment for size seems to do it.

I've tested so far on sn32 and split over UART works, even on high baud rates with no issues. I can't imagine the timeout adjustment breaking other chips with bigger FIFOs, since they should always return success faster.

File: quantum/split_common/split_util.c (hunk 1 above)
Comment (zvecr) [reply quantum/split_common/split_util.c:71]: > This should be reverted -- we want it to exit early when connected. Your changes unconditionally continue to call the delay.

I 2nd this. Now if someone is only using a single half of the keyboard they encounter a delay. One that would effect things like getting into bios.

**Comment by KarlK90** [comment]
> since the `#define`'d timeout period is shorter than the time needed for the data to be sent if the transfer is big enough. A small adjustment for size seems to do it.

For the default baudrate of 230400 Baud/s @ 8O2 and a standard timeout of 20ms you can transfer 384 bytes of data. The largest regular transfers at this point are probably the keyboard matrices which seldom exceed a couple of bytes. I'm fairly sure that the root cause is in the low level driver, perhaps not signaling that there is empty space in the tx fifo/register?

**Pull request closed without merging**